core/state/statedb.go

```diff
@@ -372,12 +372,16 @@ func (self *StateDB) deleteStateObject(stateObject *StateObject) {
 // Retrieve a state object given my the address. Returns nil if not found.
 func (self *StateDB) GetStateObject(addr common.Address) (stateObject *StateObject) {
 	// Prefer 'live' objects.
+	self.lock.Lock()
```

Comment:
Interesting that you mention it -- I drafted exactly such a thing, then read the below and changed my mind.

https://stackoverflow.com/questions/11063473/map-with-concurrent-access
> RW locks are good for resources with a lot of contention, but they have more overhead than a mutex. Map get/sets are fast enough that the program likely won't have enough contention to make the more complex lock give better throughput than a simple mutex. – rmmh Nov 11 '12 at 22:02

Would be very interested to learn more.

Also, @odeke-em, can you see the reason we're "getting away with" using maps for `stateObjects`, `stateObjectsDirty`, and `logs` _without_ any kind of locking mechanism for their read/writes in this file?